src/linux/write_zeroes.rs

```diff
@@ -32,150 +35,276 @@ impl PunchHole for File {
 
 /// A trait for writing zeroes to a stream.
 pub trait WriteZeroes {
-    /// Write zeroes to a stream.
-    ///
-    /// Write `length` bytes of zeroes to the stream, returning how many
-    /// bytes were written.
+    /// Write up to `length` bytes of zeroes to the stream, returning how many bytes were written.
     fn write_zeroes(&mut self, length: usize) -> Result<usize>;
+
+    /// Write zeroes to the stream until `length` bytes have been written.
+    ///
+    /// This method will continuously call `write_zeroes` until the requested `length` is satisfied
+    /// or an error is encountered.
+    fn write_zeroes_all(&mut self, mut length: usize) -> Result<()> {
```

Comment:
This function name sounds a bit weird in English. How about we name it `write_all_zeroes`? This way it also is closer to the `write_all` nomenclature that seem like we want to follow here. Same goes for `write_zeroes_all_at`. I believe it sounds better as `write_all_zeroes_at`.

Reply:
Yep... I chose to keep the same name thinking that `write zeroes` is the action and would be easier to spot the difference between those function and the ones without `all`. Anyway, I updated with the english-friendly name.